make-zip

```diff
@@ -1,80 +1,183 @@
+    usercode_dir: Path,
+    output_file: Path,
+    wifi_psk: str,
+    wifi_region: str,
+) -> None:
+    """
+    Create a usercode bundle for use with SR2021+ kit.
+
+    :param team_tla: TLA of the team that the zip is created for.
+    :param usercode_dir: Directory of usercode, containing a robot.py entrypoint.
+    :param output_file: File to write the zip data to, usually robot.zip.
+    :param wifi_psk: WiFi PSK used to secure robot hotspot.
+    :param wifi_region: Region that the WiFi hotspot will be used in.
+    :raise UnableToCreateUsercodeBundleException: An error occurred during creation.
+    """
+    entrypoint_path = usercode_dir / ENTRYPOINT
+
+    # Validate arguments
+    if not usercode_dir.exists() or not usercode_dir.is_dir():
+        raise UnableToCreateUsercodeBundleException(
+            f"Usercode directory ({usercode_dir}) is not a directory or doesn't exist.",
+        )
+
+    if not entrypoint_path.exists():
+        raise UnableToCreateUsercodeBundleException(
+            f"Usercode directory ({usercode_dir}) does not contain a {ENTRYPOINT} file.",
+        )
+
+    if len(team_tla) == 0:
+        raise UnableToCreateUsercodeBundleException("Team TLA cannot be empty.")
+
+    # Create the ZIP
+    with ZipFile(output_file, "w", compression=ZIP_DEFLATED) as zip_file:
+        # Add the usercode
+        for folderName, subfolders, filenames in os.walk(usercode_dir):
+            relativeFolder = os.path.relpath(folderName, usercode_dir)
+            for filename in filenames:
+                relative_path = os.path.join(relativeFolder, filename)
+                real_path = os.path.join(folderName, filename)
+
+                # Check that the file should not be ignored
+                if not is_file_ignored(relative_path):
+                    zip_file.write(real_path, relative_path)
+
+        # Add bundle.toml
+        bundle_data = get_bundle_dict(
+            "0.0.0.0dev",  # This should eventually be fetched from an API
+            f"robot-{team_tla}",
+            wifi_psk,
+            wifi_region,
+        )
+        bundle_str = toml.dumps(bundle_data)
+        zip_file.writestr("bundle.toml", bundle_str)
+
+
+def main() -> None:
+    """Pack a robot.zip from user code."""
+    parser = argparse.ArgumentParser(description="Pack a robot.zip from user code.")
+    parser.add_argument(
+        "team_tla",
+        nargs="?",
+        default="SRZ",  # Default to SRZ if no TLA is provided
+        help="The TLA of the team the zip is being created for",
+    )
+    parser.add_argument(
+        "usercode",
+        type=Path,
+        help="user code source directory, containing a robot.py",
+    )
+    parser.add_argument(
+        "output",
+        nargs="?",
+        type=Path,
+        help="output zip file",
+        default="robot.zip",
+    )
+    parser.add_argument(
+        "-p",
+        "--psk",
+        default="beeeeees",
+        help="WiFi PSK used to secure robot hotspot",
+    )
```

Comment:
The IDE currently has no mechanism to specify this. Instead the expectation is that the tool will inspect a file (`/etc/sr/wifi-keys.yaml`, though this was previously configurable) and use the value from there.

I'm not sure if that's a _good_ way to do it, though it is what the previous script did.

As this is ostensibly a secret value it's probably a good idea not to pass it directly on the command line. The previous reference path mechanism worked ok, though an environment based approach could also work.

Anything other than having the tool figure this out for itself will need work in the IDE, so if we do want to change this we may as well work out both what other changes we want to make to this interface are (I have some thoughts on that; the first being making the IDE better configurable in terms of the command being run) and design a more idealised interface rather than just adding this one thing.